### PR TITLE
[WIP] Introduce Elytron Security Filesystem

### LINKED ISSUE #624: Add support for the Elytron FileSystemSecurityRealm
In chatting with Darran, the most functional local/file based security realm store in Elytron is the org.wildfly.security.auth.realm.FileSystemSecurityRealm, so that is something the security extension needs to support and probably should be the "default".

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #624

### CODE REVIEW

**Comment by gastaldi** [comment]
Filesystem documentation for posterity: https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_elytron/components/Filesystem_Security_Realm.adoc

File: extensions/elytron-security-filesystem/deployment/src/test/java/io/quarkus/security/test/BasicAuthTestCase.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    @TempDir
+    static Path tempDir;
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(TestSecureServlet.class, TestApplication.class,
+                            RolesEndpointClassLevel.class, ParametrizedPathsResource.class,
+                            SubjectExposingResource.class)
+                    .addAsResource(new StringAsset("quarkus.security.filesystem.path=" + tempDir),
```
Comment (dmlloyd) [inline extensions/elytron-security-filesystem/deployment/src/test/java/io/quarkus/security/test/BasicAuthTestCase.java]: It looks like the path name will need to be escaped here.

**Review by gsmet — CHANGES_REQUESTED** [review]
I'm pretty sure it will require some doc updates? IIRC there's a list of Elytron extensions somewhere in the doc.

**Comment by sberyozkin** [comment]
Hi @gastaldi @geoand sorry for a delay, back from a trip home.

@gastaldi IMHO this is also very neatly done similarly to the proposed LDAP extension.
But in this case I have a small doubt.
We already have `elytron-security-properties-file` with the configuration like [this one](https://github.com/quarkusio/quarkus/blob/master/integration-tests/elytron-security/src/main/resources/META-INF/microprofile-config.properties).
This PR supports what looks like the same type of configuration but in the external XML file, is it correct ?

**Comment by gastaldi** [comment]
> This PR supports what looks like the same type of configuration but in the external XML file, is it correct ?

Yes, that's correct

**Comment by sberyozkin** [comment]
@gastaldi OK thanks. I suppose it fits with the current strategy for all the elytron extensions. My minor concern was, that the properties file is also on the filesystem and XML file sets the properties too :-) so it is not obvious what the difference is between `elytron-security-properties-file` and `elytron-security-filesystem`.
I don't really mind, but perhaps an alternative name for this extension could be considered ?

**Comment by gastaldi** [comment]
> I don't really mind, but perhaps an alternative name for this extension could be considered ?

I'm not opposed to it, but do you have anything in mind? I'm not really good at naming things :)

**Comment by sberyozkin** [comment]
@gastaldi may be `quarkus-security-filesystem-identity` (because the root XML element is `identity`) ? Or may be `quarkus-security-identity-xml`
Not sure it could confuse the elytron users who already know about `filesystem`. Let me ask Stuart @stuartwdouglas opinion as well. Guillaume @gsmet always have some name ideas too :-)
If you decide not to rename then I won't mind, thanks

**Review by sberyozkin — APPROVED** [review]

**Comment by sberyozkin** [comment]
@gastaldi Hi, if it is not `WIP` any longer then please remove it

**Comment by gastaldi** [comment]
@sberyozkin I am working in the documentation for this extension (guide).
I'll remove the WIP once I push my changes

Em Ter, 14 de jan de 2020 07:04, sberyozkin <notifications@github.com>
escreveu:

**Comment by sberyozkin** [comment]
@gastaldi OK, thanks, I thought it was ready for a merge

**Comment by sberyozkin** [comment]
@gastaldi Hi George, is it ready to go ? Can you please resolve the conflicts ?

**Comment by gastaldi** [comment]
@sberyozkin hey! I rebased the conflicts. The only thing missing is a guide/documentation for this extension.  I haven't been able to write that down yet, would you like to take a stab and add it to my PR?

**Comment by gastaldi** [comment]
Closing this for now

**Pull request closed without merging**